cursor/encoder.go

```diff
@@ -47,7 +47,11 @@ func (e *Encoder) marshalJSON(model interface{}) ([]byte, error) {
 		} else {
 			// fetch values from custom types
 			if ct, ok := util.ReflectValue(f).Interface().(CustomType); ok {
-				fields[i] = ct.GetCustomTypeValue(field.Meta)
+				value, err := ct.GetCustomTypeValue(field.Meta)
+				if err != nil {
+					return nil, err
```

Comment:
I was considering if we should return another error here (like we do with `ErrInvalidModel`) instead of just propagating the error from `GetCustomTypeValue`, but decided to just propagate the error. WDYT?

Reply:
Another consideration is that `GetCustomTypeValue` would be implemented by clients, and if we convert their error implicitly here, we might surprise them.

But both ways make sense to me. You can choose the comfortable one and I would support your decision 💪

Reply:
Right. Let's keep it as is and just propagate the error.

Reply:
No problem 🙌